tidy3d/components/device/simulation.py

```diff
@@ -220,9 +230,63 @@ def names_exist_sources(cls, val, values):
                     )
         return val
 
+    @pd.validator("boundary_spec", always=True)
+    @skip_if_fields_missing(["structures", "medium"])
+    def unique_type_bc(cls, val, values):
+        """Error if boundary specs for different type of simulations are provided."""
+        # NOTE: this must be removed once we're able to deal with coupled simulations
+
+        types = []
+        for bc in val:
+            if type(bc.condition) in (TemperatureBC, HeatFluxBC, ConvectionBC):
+                t = DeviceSimulationType.HEAT
+                if t not in types:
+                    types.append(t)
+            elif type(bc.condition) in (PotentialBC, InsulatingBC):
+                t = DeviceSimulationType.CONDUCTION
+                if t not in types:
+                    types.append(t)
+
+        if len(types) > 1:
+            raise SetupError(
+                "Boundary conditions for different types of simulation "
+                "have been provided, i.e., HEAT and CONDUCTION type BCs have been mixed."
+            )
+        return val
+
+    @pd.validator("boundary_spec", always=True)
+    @skip_if_fields_missing(["sources", "structures", "medium"])
+    def source_and_bcs_consistent(cls, val, values):
```

Comment:
we should also make sure that monitors are consistent with simulation type. Given that there are many components needs to be checked, maybe it makes sense to implement this as a root validator that checks the entire model, for example, https://github.com/flexcompute/tidy3d/blob/develop/tidy3d/components/medium.py#L1566